stratosphere/ams_mitm/source/utils.cpp

```diff
@@ -665,3 +667,88 @@ Result Utils::GetSettingsItemBooleanValue(const char *name, const char *key, boo
 void Utils::RebootToFatalError(AtmosphereFatalErrorContext *ctx) {
     BpcRebootManager::RebootForFatalError(ctx);
 }
+
+void Utils::CreateBlankProdinfoIfNeeded() {
+    /* if (!Utils::HasGlobalFlag("blank_prodinfo")) {
+        return;
+    } */
+
+    Result rc;
+
+    u8* buffer = g_cal0_backup;
+
+    // big brain: modify buffer inline
+
+    const char* serial = "XAW00000000000"; // TODO: Make this configurable?
+
+    memcpy(&buffer[0x250], serial, strlen(serial));
+
+    const u64 erasePoints[14] = {0xAE0, 0x800, 0x3AE0, 0x130, 0x35E1, 0x6, 0x36E1, 0x6, 0x2B0, 0x180, 0x3D70, 0x240, 0x3FC0, 0x240}; // offset, size
```

Comment:
Please do not store buffers that alternate different values. It is really bad style.

Reply:
What do you mean by alternate different values?

Reply:
You're storing two conceptually different things (offset, value) in a single buffer.

Also can mark this static constexpr.

Reply:
Why constexpr over const?

Reply:
const will actually initialize the buffer on stack.

marking it static will move it into .rodata

static constexpr is preferable over static const because it lets the compiler know it's knowable at compiltime.